apps/ex_wire/lib/ex_wire/struct/routing_table.ex

```diff
@@ -0,0 +1,156 @@
+defmodule ExWire.Struct.RoutingTable do
+  @moduledoc """
+  Module for working with current node's buckets
+  """
+
+  alias ExWire.Struct.{Bucket, Peer}
+  alias ExWire.KademliaConfig
+
+  @network Application.fetch_env(:ex_wire, :network_process_name)
+
+  defstruct [:current_node, :buckets]
+
+  @type t :: %__MODULE__{
+          current_node: Peer.t(),
+          buckets: [Bucket.t()]
+        }
+
+  @doc """
+  Creates new routing table.
+
+  ## Examples
+
+      iex> node = ExWire.Struct.Peer.new("13.84.180.240", 30303, "6ce05930c72abc632c58e2e4324f7c7ea478cec0ed4fa2528982cf34483094e9cbc9216e7aa349691242576d552a2a56aaeae426c5303ded677ce455ba1acd9d", time: :test)
+      iex> table = node |> ExWire.Struct.RoutingTable.new()
+      iex> table.buckets |> Enum.count
+      256
+  """
+  @spec new(Peer.t()) :: t()
+  def new(peer = %Peer{}) do
+    initial_buckets = initialize_buckets()
+
+    %__MODULE__{
+      current_node: peer,
+      buckets: initial_buckets
+    }
+  end
+
+  @doc """
+  Returns table's buckets.
+  """
+  @spec buckets(t()) :: [Bucket.t()]
+  def buckets(%__MODULE__{buckets: buckets}), do: buckets
+
+  @doc """
+  Adds node to routing table.
+  """
+  @spec refresh_node(t(), Peer.t()) :: t()
+  def refresh_node(
+        table = %__MODULE__{current_node: %Peer{remote_id: current_node_id}},
+        %Peer{remote_id: current_node_id}
+      ),
+      do: table
+
+  def refresh_node(table = %__MODULE__{buckets: buckets}, node = %Peer{}) do
+    bucket_idx = bucket_id(table, node)
+
+    refresh_node_result =
+      buckets
+      |> Enum.at(bucket_idx)
+      |> Bucket.refresh_node(node)
+
+    case refresh_node_result do
+      {:full_bucket, candidate_for_removal, bucket} ->
+        handle_full_bucket(table, bucket, candidate_for_removal, node)
+        table
+
+      {_descr, _node, bucket} ->
+        replace_bucket(table, bucket_idx, bucket)
+    end
+  end
+
+  @doc """
+  Returns neighbours of a specified node.
+  """
+  @spec neighbours(t(), Peer.t()) :: [Peer.t()]
+  def neighbours(table = %__MODULE__{}, node = %Peer{}) do
+    bucket_idx = bucket_id(table, node)
+    similar_to_current_node = nodes_at(table, bucket_idx)
```

Comment:
is `similar` the same as `near`? I want to avoid using two different terms to describe the same concept if possible. `nearest_neighbors` maybe?